File: 判断上三角矩阵.py
T = int(input())
for i in range(T):
    n = int(input())
    a = []
    for j in range(n):
        a.append(list(map(int,input().split())))
    flag = 0
    print(a)
    for k in range(n):
        for h in range(n):
            if (k>h) and (a[k][h]!=0):
                print("NO")
                flag = 1
                break
        if flag == 1:
            break
    else:
        print("YES")
        

File: 判断上三角矩阵.c
#include<stdio.h>
#include<string.h>
int main()
{
   int T = 0;
   scanf("%d",&T);
   for(int k = 0;k < T;k++)
   {    int n = 0;
        scanf("%d",&n);
        int a[n][n];
        int flag = 1;
        for(int i = 0;i < n;i++)
        {
            for(int j = 0;j < n;j++)
            {
                scanf("%d",&a[i][j]);
                if(i>j&&a[i][j]!=0)
                {
                    flag = 0;
                }
            }
        }
        if(flag==0)
            printf("NO\n");
        else
            printf("YES\n");
   }
}
